### ISSUE TITLE
spec language of calling ThrowCompletion seems strange

### ISSUE THREAD

**Issue by FrankYFTang** [tc39/proposal-intl-messageformat#65]
I think the language used in the spec text about calling ThrowCompletion() may need to be change via editorial PR

We now have

1. Let error be ThrowCompletion(TypeError).
in https://tc39.es/proposal-intl-messageformat/#sec-selectpattern

a. Let error be ThrowCompletion(ReferenceError).
in https://tc39.es/proposal-intl-messageformat/#sec-resolveexpression

a. Let error be ThrowCompletion(ReferenceError).
i. Let error be ThrowCompletion(TypeError).
in https://tc39.es/proposal-intl-messageformat/#sec-resolvefunction

i. Let error be ThrowCompletion(ReferenceError).
1. Let error be ThrowCompletion(ReferenceError).
in https://tc39.es/proposal-intl-messageformat/#sec-resolvevalue

but ThrowCompletion is expecting a an [ECMAScript language value](https://tc39.es/ecma262/#sec-ecmascript-language-types)
see https://tc39.es/ecma262/#sec-throwcompletion
but TypeError and ReferenceError are "Global Name" as defined in  
and https://tc39.es/ecma262/#sec-well-known-intrinsic-objects

Is "Global Name" an [ECMAScript language value](https://tc39.es/ecma262/#sec-ecmascript-language-types) ? 

Looks other part of ecma262. it seems the language in spec text is

1. Return [ThrowCompletion](https://tc39.es/ecma262/#sec-throwcompletion)(a newly created TypeError object).

i. Let error be [ThrowCompletion](https://tc39.es/ecma262/#sec-throwcompletion)(a newly created TypeError object).

a. Let error be a newly created TypeError object.

1. Let error be a newly created AggregateError object.
etc

**Comment by ljharb** [comment]
Indeed, `ThrowCompletion` does not take a constructor, it takes an error object.
